app/controllers/image_controller.rb

```diff
@@ -746,17 +746,17 @@ def license_updater
     process_license_changes if request.method == "POST"
 
     # Gather data for form.
-    @data = Image.connection.select_all(%(
-      SELECT COUNT(*) AS license_count, copyright_holder, license_id
-      FROM images
-      WHERE user_id = #{@user.id.to_i}
-      GROUP BY copyright_holder, license_id
-    )).to_a
-    @data.each do |datum|
+    # map(&:attributes) gives you a hash of your selects with their keys
+    @data = Image.includes(:license).where(user_id: @user.id).
+            select(Arel.star.count.as("license_count"),
+                   :copyright_holder, :license_id).
+            group(:copyright_holder, :license_id).
+            map(&:attributes).map do |datum|
       next unless (license = License.safe_find(datum["license_id"].to_i))
```

Comment:
Further research: 
- `as_json` gets the same result as `map(&:attributes)`, but...
- `pluck` and manually mapping the results to keys seems like it will be definitely faster than this select, considering we're talking about a lot of images. 
- The popular and charmingly named gem `pluck_all`, that i had wanted to install during the initial SQL-to-AR sweep, does it in one go. So does the gem `pluck_to_hash`, but `pluck_all` is actively maintained. Pending votes on the gem, i'm converting this to `pluck` now.

Reply:
Works for me, thanks, Nimmo.

Reply:
Going to keep it at `pluck`, without the gem. 
We are manually adding to the hash right afterwards, so it seems cleaner to spec the first three keys manually too.

Reply:
Is there any functional difference between these Ruby hashes? Generated by the original SQL:
```ruby
[{"license_count"=>10764,                                                       
  "copyright_holder"=>"Jason Hollinger",                                        
  "license_id"=>3,                                                              
  "license_name"=>"Creative Commons Wikipedia Compatible v3.0",
  "licenses"=>[["Creative Commons Non-commercial v3.0", 2], ["Creative Commons Wikipedia Compatible v3.0", 3], ["Public Domain", 4]]},
 {"license_count"=>4,               
  "copyright_holder"=>"Alan Cressler",
  "license_id"=>3,                  
  "license_name"=>"Creative Commons Wikipedia Compatible v3.0",
  "licenses"=>[["Creative Commons Non-commercial v3.0", 2], ["Creative Commons Wikipedia Compatible v3.0", 3], ["Public Domain", 4]]},
 {"license_count"=>1,
  "copyright_holder"=>"Tim Wheeler",
  "license_id"=>2,
  "license_name"=>"Creative Commons Non-commercial v3.0",
  "licenses"=>[["Creative Commons Non-commercial v3.0", 2], ["Creative Commons Wikipedia Compatible v3.0", 3], ["Public Domain", 4]]}] 
```
Generated by the `pluck` and `map`:
```ruby
[{:license_count=>10764,                                                       
  :copyright_holder=>"Jason Hollinger",                                        
  :license_id=>3,                                                              
  :license_name=>"Creative Commons Wikipedia Compatible v3.0",
  :licenses=>[["Creative Commons Non-commercial v3.0", 2], ["Creative Commons Wikipedia Compatible v3.0", 3], ["Public Domain", 4]]},
 {:license_count=>4,               
  :copyright_holder=>"Alan Cressler",
  :license_id=>3,                  
  :license_name=>"Creative Commons Wikipedia Compatible v3.0",
  :licenses=>[["Creative Commons Non-commercial v3.0", 2], ["Creative Commons Wikipedia Compatible v3.0", 3], ["Public Domain", 4]]},
 {:license_count=>1,
  :copyright_holder=>"Tim Wheeler",
  :license_id=>2,
  :license_name=>"Creative Commons Non-commercial v3.0",
  :licenses=>[["Creative Commons Non-commercial v3.0", 2], ["Creative Commons Wikipedia Compatible v3.0", 3], ["Public Domain", 4]]}] 
```

Reply:
Oh! I have found one difference.  Ruby is SO WEIRD about strings and symbols... sometimes it converts, sometimes not.
You can't call  `datum["license_count"]` on the second hash. 
You MUST call `datum[:license_count]`

Reply:
This is the kind of thing where i start to wonder if i'm learning useful stuff, with Ruby, or just getting to know a really quirky language. I guess all languages have their quirks. Javascript more than most.